modules/apps/site/site-cms-site-initializer/src/main/java/com/liferay/site/cms/site/initializer/internal/servlet/DownloadObjectEntryFolderServlet.java

```diff
@@ -306,6 +258,145 @@ private String _getObjectEntryFolderPath(
 		return requestURI.substring(path.length() + 1);
 	}
 
+	private User _getUser(HttpServletRequest httpServletRequest)
+		throws PortalException {
+
+		User user = _portal.getUser(httpServletRequest);
+
+		if (user == null) {
+			user = _userLocalService.getGuestUser(
+				_portal.getCompanyId(httpServletRequest));
+		}
+
+		return user;
+	}
+
+	private void _handleSelectAllDownload(
+			ThemeDisplay themeDisplay, User user, ZipWriter zipWriter)
+		throws IOException, PortalException {
+
+		BooleanQuery booleanQuery = _queries.booleanQuery();
+
+		booleanQuery.addFilterQueryClauses(_queries.term("cms_kind", "object"));
+
+		TermsQuery cmsSection = _queries.terms("cms_section");
+
+		cmsSection.addValues("contents", "files");
+
+		booleanQuery.addFilterQueryClauses(cmsSection);
+
+		SearchResponse searchResponse = _searcher.search(
+			_searchRequestBuilderFactory.builder(
+			).emptySearchEnabled(
+				true
+			).fetchSource(
+				true
+			).query(
+				booleanQuery
+			).withSearchContext(
+				searchContext -> {
+					searchContext.setCompanyId(user.getCompanyId());
+					searchContext.setEnd(QueryUtil.ALL_POS);
+					searchContext.setLocale(user.getLocale());
+					searchContext.setStart(QueryUtil.ALL_POS);
+					searchContext.setTimeZone(user.getTimeZone());
+					searchContext.setUserId(user.getUserId());
+				}
+			).build());
+
+		SearchHits searchHits = searchResponse.getSearchHits();
+
+		for (SearchHit searchHit : searchHits.getSearchHits()) {
+			Document document = searchHit.getDocument();
+
+			if (StringUtil.equalsIgnoreCase(
+					document.getString("entryClassName"),
+					_BASIC_DOCUMENT_MODEL_CLASS_NAME)) {
+
+				_zipObjectEntry(
```

Comment:
why not managing folders here, like when we get the IDs list?

Reply:
> why not managing folders here, like when we get the IDs list?

The reason is that folders are not returned by search as in the All section of the CMS